x-pack/plugins/observability/public/components/shared/slo/slo_selector/slo_selector.test.tsx

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { act, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { render } from '../../../../utils/test_helper';
+import { SloSelector } from './slo_selector';
+import { useFetchSloList } from '../../../../hooks/slo/use_fetch_slo_list';
+import { emptySloList } from '../../../../pages/slos/mocks/slo_list';
+import { wait } from '@testing-library/user-event/dist/utils';
+
+jest.mock('../../../../hooks/slo/use_fetch_slo_list');
+const useFetchSloListMock = useFetchSloList as jest.Mock;
+
+describe('SLO Selector', () => {
+  const onSelectedSpy = jest.fn();
+  beforeEach(() => {
+    jest.clearAllMocks();
+    useFetchSloListMock.mockReturnValue([true, emptySloList]);
+  });
+
+  it('fetches SLOs asynchronously', async () => {
+    render(<SloSelector onSelected={onSelectedSpy} />);
+
+    expect(screen.getByTestId('sloSelector')).toBeTruthy();
+    expect(useFetchSloListMock).toHaveBeenCalledWith('');
+  });
+
+  it('searches SLOs when typing', async () => {
+    const { getByTestId } = render(<SloSelector onSelected={onSelectedSpy} />);
+
+    const input = getByTestId('comboBoxInput');
+    await act(async () => {
```

Comment:
You may not need `act`: https://flyingsquirrel.medium.com/when-should-i-use-act-in-react-testing-library-d7dd22a3340e

Reply:
Thanks for the link, going to read this and fix the test accordingly 👍🏻

Reply:
Cannot make it work without act since the hook is called after some delays after the typing.